datatree/tests/test_datatree.py

```diff
@@ -693,3 +693,42 @@ def test_filter(self):
         )
         elders = simpsons.filter(lambda node: node["age"] > 18)
         dtt.assert_identical(elders, expected)
+
+
+class TestSubtreeMerge:
+    def test_merge_subtrees(self):
+        data_tree = xr.DataArray(
+            np.random.rand(3, 3, 3),
+            coords=[("x", [0, 1, 2]), ("y", [0, 1, 2]), ("z", [0, 1, 2])],
+            dims=["x", "y", "z"],
+        ).groupby("x")
+        merged_data_tree = data_tree.merge_subtrees(subtree_a=0, subtree_b=1)
```

Comment:
Because your `data_tree` object is not actually an instance of the class `DataTree`, your new method `merge_subtrees` is not defined on this object, which is why [the continuous integration tests fail](https://github.com/xarray-contrib/datatree/actions/runs/4469070340/jobs/7863573646?pr=227) with an `AttributeError` here.

Did you try running the tests locally first? They would have failed locally too.